addon/manifests/chart/templates/deployment.yaml

```diff
@@ -84,8 +84,21 @@ spec:
           limits:
             memory: 1024Mi
           {{- end }}
+        {{- if .Values.prometheus.enabled }}
```

Comment:
I don't think we need this if condition.
liveness and readiness endpoints are provided by the image. It is independent of prometheus.